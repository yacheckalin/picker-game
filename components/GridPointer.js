import React from "react";
import PropTypes from "prop-types";

import styled from "styled-components";

const StyledPointer = styled.div`
  box-sizing: border-box;
  width: ${(props) => props.width}px;
  background-color: yellowgreen;
  height: ${(props) => props.height}px;
  position: absolute;
  top: ${(prop) => prop.top}px;
  left: ${(prop) => prop.left}px;
  opacity: 0.5;

  i {
    width: ${(props) => props.width}px;
    font-size: 2em;
  }
`;

const GridPointer = ({ width, height, top, left }) => (
  <>
    <StyledPointer
      width={width}
      height={height}
      left={left}
      top={top}
      className="valign-wrapper"
    >
      <i className="material-icons center-align">flare</i>
    </StyledPointer>
  </>
);

GridPointer.propTypes = {
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  top: PropTypes.number.isRequired,
  left: PropTypes.number.isRequired,
};

export default GridPointer;
