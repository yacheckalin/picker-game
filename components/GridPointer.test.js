import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import GridPointer from "./GridPointer";

describe("<GridPointer />", () => {
  test("render and displays properly", () => {
    const props = { width: 100, height: 100, left: 0, top: 0 };
    const wrapper = shallow(<GridPointer {...props} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
