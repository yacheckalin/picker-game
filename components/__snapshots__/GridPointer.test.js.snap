// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridPointer /> render and displays properly 1`] = `
<styled.div
  className="valign-wrapper"
  height={100}
  left={0}
  top={0}
  width={100}
>
  <i
    className="material-icons center-align"
  >
    flare
  </i>
</styled.div>
`;
